src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.constructor.alive.push(this);
+  }
+
+  static getAlive() {
+    return this.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (!prey.hidden && prey instanceof Herbivore) {
+      prey.health -= 50;
+
+      if (prey.health <= 0) {
+        prey.healtth = 0;
+
+        const index = Animal.alive.indexOf(prey);
+
+        if (index > -1) {
+          Animal.alive.splice(index, 1);
+        }
```

Comment:
[CODE PERFORMANCE] Instead of using indexOf and splice to remove dead animals, consider using `filter` to avoid nesting loops and improve performance.